src/main/java/fence/parser/Parser.java

```diff
@@ -0,0 +1,177 @@
+package fence.parser;
+
+import fence.task.Deadline;
+import fence.task.Event;
+import fence.task.Task;
+import fence.task.Todo;
+import java.time.LocalDate;
+import java.util.NoSuchElementException;
+import java.util.StringTokenizer;
```

Comment:
The use of StringTokenizer to parse commands is a clever choice. However, it might be a bit hard to follow for those who are not familiar with StringTokenizer.